File: src/main/java/it/unibo/crabinv/Model/audio/BGMTracks.java
package it.unibo.crabinv.Model.audio;

/**
 * Provides a set list of present BGM Tracks and their relative path in resources
 */
public enum BGMTracks {
    LEVEL("/audio/BGM/main_bgm.mp3"),
    MENU("/audio/BGM/menu_bgm.mp3"),;

    private final String path;

    BGMTracks(String path) {
        this.path = path;
    }

    /**
     * @return the resource path of the selected BGM
     */
    public String getPath() {
        return path;
    }
}

File: src/main/java/it/unibo/crabinv/model/core/audio/BGMTracks.kt
package it.unibo.crabinv.model.core.audio

/**
 * Provides a set list of present BGM Tracks and their relative path in resources.
 */
enum class BGMTracks(
    /**
     * @return the resource path of the selected BGM
     */
    val path: String,
) {
    LEVEL("/audio/BGM/main_bgm.mp3"),
    MENU("/audio/BGM/menu_bgm.mp3"),
}
